File: testers/avro/java-quarkus/generated-and-checked-in/com/example/events/LinkedListNodePublisher.java
package com.example.events;

import io.smallrye.mutiny.Uni;
import io.smallrye.reactive.messaging.MutinyEmitter;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import org.eclipse.microprofile.reactive.messaging.Channel;

/** Type-safe event publisher for linked-list-node topic */
@ApplicationScoped
public record LinkedListNodePublisher(
    @Channel("linked-list-node") MutinyEmitter<LinkedListNode> kafkaTemplate, String topic) {
  @Inject
  public LinkedListNodePublisher {}

  public LinkedListNodePublisher(
      @Channel("linked-list-node") MutinyEmitter<LinkedListNode> kafkaTemplate) {
    this(kafkaTemplate, "linked-list-node");
  }

  public LinkedListNodePublisher withKafkaTemplate(MutinyEmitter<LinkedListNode> kafkaTemplate) {
    return new LinkedListNodePublisher(kafkaTemplate, topic);
  }

  public LinkedListNodePublisher withTopic(String topic) {
    return new LinkedListNodePublisher(kafkaTemplate, topic);
  }

  /** Publish a LinkedListNode event */
  public Uni<Void> publish(String key, LinkedListNode event) {
    return kafkaTemplate.send(event);
  }
}

File: testers/avro/kotlin-quarkus-mutiny/generated-and-checked-in/com/example/events/LinkedListNodePublisher.kt
package com.example.events

import io.smallrye.mutiny.Uni
import io.smallrye.reactive.messaging.MutinyEmitter
import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import java.lang.Void
import org.eclipse.microprofile.reactive.messaging.Channel

@ApplicationScoped
/** Type-safe event publisher for linked-list-node topic */
data class LinkedListNodePublisher @Inject constructor(
  @field:Channel("linked-list-node") val kafkaTemplate: MutinyEmitter<LinkedListNode>,
  val topic: kotlin.String = "linked-list-node"
) {
  /** Publish a LinkedListNode event */
  fun publish(
    key: kotlin.String,
    event: LinkedListNode
  ): Uni<Void> {
    return kafkaTemplate.send(event)
  }
}